modules/manifold/src/components/ui/segment-filters-control/segment-panel.js

```diff
@@ -0,0 +1,163 @@
+  background-color: ${props =>
+    props.isOpen ? '#fff' : THEME.colors.inputFill};
+  font-size: 13px;
+  font-weight: normal;
+  color: #000;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  height: 36px;
+  padding: 4px 4px 4px 8px;
+`;
+
+const HeaderTextGroup = styled.div`
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+  padding-right: 8px;
+`;
+
+const IconButton = styled.div`
+  width: 24px;
+  height: 24px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
+  color: ${props => (props.disabled ? '#ccc' : '#000')}
+  margin: 1px;
+  transition: 0.2s;
+`;
+
+export default class SegmentPanel extends PureComponent {
+  static propTypes = {
+    segmentId: PropTypes.number.isRequired,
+    nSegments: PropTypes.number.isRequired,
+    filters: PropTypes.arrayOf(FILTER),
+    columnDefs: PropTypes.arrayOf(FIELD),
+    onRemoveSegment: PropTypes.func,
+    onUpdateFilterValue: PropTypes.func,
+  };
+
+  static defaultProps = {
+    filters: [],
+    columnDefs: [],
+    onRemoveSegment: () => {},
+    onUpdateFilterValue: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {isOpen: props.segmentId === 0};
+  }
+
+  _onToggleOpen = () => {
+    this.setState({
+      isOpen: !this.state.isOpen,
+    });
+  };
+
+  _onRemoveSegment = segmentId => {
+    if (this.props.nSegments > 2) {
+      this.props.removeSegment(segmentId);
+    }
+  };
+
+  _handleAddRemoveFilters = newKeys => {
+    const {
+      attributes,
+      segment: {filters = []},
+    } = this.props;
+
+    const oldKeys = filters.map(f => f.key);
+    const keysToAppend = newKeys.filter(key => !oldKeys.includes(key));
+    const keysToRemove = oldKeys.filter(key => !newKeys.includes(key));
+
+    const filtersToAppend = keysToAppend.map(key => {
+      const {type: attributeType} = attributes.find(attr => attr.key === key);
+      return {
+        key,
+        keyIndex: 0,
+        type: FEATURE_TYPE_TO_FILTER_TYPE_MAP[attributeType],
+      };
+    });
+
+    const updatedFilters = filters
+      .filter(f => !keysToRemove.includes(f.key))
+      .concat(filtersToAppend)
+      .sort((a, b) => a.key.localeCompare(b.key));
+
+    this.props.updateFilters(updatedFilters);
+  };
+
+  render() {
+    const {
+      nSegments,
+      segmentId,
+      filters,
+      columnDefs,
+      onRemoveSegment,
+      onUpdateFilterValue,
+    } = this.props;
+    const {isOpen} = this.state;
+
+    return (
+      <Container isOpen={isOpen}>
+        <Header isOpen={isOpen}>
+          <HeaderTextGroup onClick={this._onToggleOpen}>
+            <IconButton>
+              {isOpen ? <ChevronDown size={22} /> : <ChevronRight size={22} />}
+            </IconButton>
+            {`segment ${segmentId}`}
+          </HeaderTextGroup>
+          <IconButton
+            disabled={nSegments < 2}
```

Comment:
From above we were checking for `nSegments > 2`.
What if `nSegments === 2` here?

Reply:
Changed to <=